### PR TITLE
Add accelerator module

### PR DESCRIPTION
**Pull request by wusyong**

### CODE REVIEW

File: src/accelerator.rs [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+            SysMods::None => RawMods::None,
+            SysMods::Shift => RawMods::Shift,
+            SysMods::Cmd => RawMods::Ctrl,
+            SysMods::AltCmd => RawMods::AltCtrl,
+            SysMods::CmdShift => RawMods::CtrlShift,
+            SysMods::AltCmdShift => RawMods::AltCtrlShift,
+        }
+    }
+}
+
```
Comment (amrbashir) [inline src/accelerator.rs:250]:
```suggestion
```

File: src/accelerator.rs [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    pub fn new(accelerator_string: &str) -> AcceleratorId {
+        AcceleratorId(hash_string_to_u16(accelerator_string))
+    }
+
+    /// Whenever this menu is empty.
+    pub fn is_empty(self) -> bool {
+        Self::EMPTY == self
+    }
+}
+
```
Comment (amrbashir) [inline src/accelerator.rs:287]: Not needed
```suggestion
```

File: src/accelerator.rs [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+    title.to_uppercase().hash(&mut s);
+    s.finish() as u16
+}
+
+fn hash_accelerator_to_u16(hotkey: Accelerator) -> u16 {
+    let mut s = DefaultHasher::new();
+    hotkey.hash(&mut s);
+    s.finish() as u16
+}
+
```
Comment (amrbashir) [inline src/accelerator.rs:378]: Not needed, just use `utill::Counter` to generate ids
```suggestion
```

File: src/accelerator.rs [hunk 4]
```diff
@@ -0,0 +1,467 @@
...
+    }
+
+    /// Returns an identifier unique to the accelerator.
+    pub fn id(self) -> AcceleratorId {
+        if let Some(id) = self.id {
+            return id;
+        }
+
+        AcceleratorId(hash_accelerator_to_u16(self))
+    }
```
Comment (amrbashir) [inline src/accelerator.rs:82]: Not needed. Accelerator is kind of a metadata, they serve nothing on their own. They don't need ids.

**Review by amrbashir — CHANGES_REQUESTED** [review]

**Comment by wusyong** [comment]
There are some doc tests failed outside of this PR.
But I think we should address it.
I'll fix it tomorrow.

File: examples/tao.rs [hunk 5]
```diff
@@ -19,7 +23,11 @@ fn main() {
 
     let mut file_menu = menu_bar.add_submenu("&File", true);
     let mut open_item = file_menu.add_item("&Open", true, None);
-    let mut save_item = file_menu.add_item("&Save", true, Some("CommandOrCtrl+S"));
+    let mut save_item = file_menu.add_item(
+        "&Save",
+        true,
+        Some(Accelerator::new(Mods::CtrlMeta, Code::KeyS)),
```
Comment (amrbashir) [inline examples/tao.rs:29]:
```suggestion
        Some(Accelerator::new(Mods::Ctrl, Code::KeyS)),
```

File: examples/winit.rs [hunk 6]
```diff
@@ -37,7 +41,11 @@ fn main() {
 
     let mut file_menu = menu_bar.add_submenu("&File", true);
     let mut open_item = file_menu.add_item("&Open", true, None);
-    let mut save_item = file_menu.add_item("&Save", true, Some("CommandOrCtrl+S"));
+    let mut save_item = file_menu.add_item(
+        "&Save",
+        true,
+        Some(Accelerator::new(Mods::CtrlMeta, Code::KeyS)),
```
Comment (amrbashir) [inline examples/winit.rs:47]:
```suggestion
        Some(Accelerator::new(Mods::Ctrl, Code::KeyS)),
```

File: src/accelerator.rs [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+        }
+    }
+
+    Ok(Accelerator { key, mods })
+}
+
+#[test]
+fn test_parse_accelerator() {
+    assert_eq!(
+        parse_accelerator("CTRL+KeyX").unwrap(),
```
Comment (amrbashir) [inline src/accelerator.rs:213]: This notation of `KeyX` will definitely break tauri so temporarily until tauri v2, we are gonna need to make a compatibility layer but not sure whether it should be in muda or tauri.
cc @lucasfernog

File: src/accelerator.rs [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+/// Represents the active modifier keys.
+///
+/// This is intended to be clearer than [`Modifiers`], when describing accelerators.
+///
+#[non_exhaustive]
+#[derive(Debug, Clone, Copy, PartialEq, Hash)]
+pub enum Mods {
+    None,
+    Alt,
+    Ctrl,
```
Comment (amrbashir) [inline src/accelerator.rs:83]: Missing variants for macos
```suggestion
    Ctrl,
	Command,
    CommandOrCtrl,
```

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: src/accelerator.rs (hunk 7 above)
Comment (wusyong) [reply src/accelerator.rs:220]: How about we just fork the types in `keyboard_types` then. I feel like it doesn't need to stick to origin crate so hard.
We only need its `Code` and `Modifiers` types.

Comment (amrbashir) [reply src/accelerator.rs:220]: No need to fork, we just need to have an internal function that takes the `Code` and converts it to a string that is compatible with current tao implementation. We only need to keep it around until tauri v2

**Review by amrbashir — APPROVED** [review]

**Pull request merged**